src/main/java/com/rsr5/gocd/riemann_notifier/PipelineDetailsPopulator.java

```diff
@@ -26,11 +41,42 @@ private JsonObject mergeInPipelineInstanceDetails(String attributeName,
         return json;
     }
 
-    protected JsonElement downloadPipelineDetails(String pipelineName) throws IOException {
-        String path = "/go/api/pipelines/" + pipelineName + "/history";
-        JsonObject json = accessor.get(path).getAsJsonObject();
-        JsonArray pipelines = json.get("pipelines").getAsJsonArray();
-        return pipelines.get(0);
+    /**
+     * Performs a depth-first search of the artifacts, returning a JsonObject
+     * translation of the first artifact named "results.xml".
+     */
+    protected JsonElement findJunitResults(JsonArray artifacts) throws IOException {
+        for (int i = 0; i < artifacts.size(); i++) {
+            JsonObject artifact = artifacts.get(i).getAsJsonObject();
+            if ("folder".equals(artifact.get("type").getAsString())) {
+                JsonElement result = findJunitResults(artifact.get("files").getAsJsonArray());
+                if (null != result) {
+                    return result;
+                }
+            } else if ("file".equals(artifact.get("type").getAsString()) &&
+                       "results.xml".equals(artifact.get("name").getAsString())) {
```

Comment:
Won't comment on any more of these, but similarly:
```suggestion
            } else if ("file" == artifact.get("type").getAsString()) &&
                       "results.xml" == artifact.get("name").getAsString())) {
```